File: exercism/rust/difference-of-squares.rs
pub fn square_of_sum(n: u64) -> u64 {
    (n * (n + 1) / 2).pow(2)
}

pub fn sum_of_squares(n: u64) -> u64 {
    n * (n + 1) * (2 * n + 1) / 6
}

pub fn difference(n: u64) -> u64 {
    square_of_sum(n) - sum_of_squares(n)
}

File: exercism/python/difference_of_squares.py
def square_of_sum(number):
    return (number * (number+1)/2)**2


def sum_of_squares(number):
    return sum(map(lambda x: x**2, range(1, number+1)))


def difference_of_squares(number):
    return square_of_sum(number) - sum_of_squares(number)
